app/src/main/java/com/github/palFinderTeam/palfinder/ProfileActivity.kt

```diff
@@ -49,9 +51,18 @@ class ProfileActivity : AppCompatActivity() {
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
+        val sharedPref = getSharedPreferences("theme", Context.MODE_PRIVATE) ?: return
+        val theme = sharedPref.getInt("theme", R.style.palFinder_default_theme)
+        setTheme(theme)
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_profile)
-
+        var sharedPreferenceChangeListener =
+            SharedPreferences.OnSharedPreferenceChangeListener { _, key ->
+                if (key == "theme") {
+                    recreate()
+                }
+            }
+        sharedPref.registerOnSharedPreferenceChangeListener(sharedPreferenceChangeListener)
```

Comment:
This seems like duplicated code across the activites. For now it's fine but might be good to refactor later!

Reply:
Unfortunately impossible, the listener need to be set here, The theme setter could be moved out but it would be very changing.